domain/application/state/storage.go

```diff
@@ -110,12 +379,98 @@ WHERE  storage_id = $storageInstance.storage_id
 	return inst.StorageUUID, nil
 }
 
+// AttachStorage attaches the specified storage to the specified unit.
+// The following error types can be expected:
+// - [github.com/juju/juju/domain/storage/errors.StorageNotFound] when the storage doesn't exist.
+// - [github.com/juju/juju/domain/application/errors.UnitNotFound]: when the unit does not exist.
+// - [github.com/juju/juju/domain/application/errors.StorageAlreadyAttached]: when the attachment already exists.
+// - [github.com/juju/juju/domain/application/errors.FilesystemAlreadyAttached]: when the filesystem is already attached.
+// - [github.com/juju/juju/domain/application/errors.VolumeAlreadyAttached]: when the volume is already attached.
+// - [github.com/juju/juju/domain/application/errors.UnitNotAlive]: when the unit is not alive.
+// - [github.com/juju/juju/domain/application/errors.StorageNotAlive]: when the storage is not alive.
+// - [github.com/juju/juju/domain/application/errors.StorageNameNotSupported]: when storage name is not defined in charm metadata.
+// - [github.com/juju/juju/domain/application/errors.InvalidStorageCount]: when the allowed attachment count would be violated.
+// - [github.com/juju/juju/domain/application/errors.InvalidStorageMountPoint]: when the filesystem being attached to the unit's machine has a mount point path conflict.
```

Comment:
Thanks for doing this, but do we need the full path?

Reply:
When first written yes because the errors package wasn't imported yet because we just had stubs. Can change now \o/